tree/30_generic_methods/augeas_commands.cf

```diff
@@ -0,0 +1,78 @@
+#####################################################################################
+# Copyright 2020 Normation SAS
+#####################################################################################
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, Version 3.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+#####################################################################################
+#
+# @name Augeas Commands
+# @description Use augeas binaries to execute augtool command directly on the agent. 
+# The generic method will return the command execution output otherwise if the augeas isn't installed on the agent, it will produces an error.
+#
+# @parameter variable_prefix The prefix of the variable name
+# @parameter variable_name   The variable to define, the full name will be variable_prefix.variable_name
+# @parameter commands        The augeas command(s) 
+# @parameter noautoload      Activate the '--no autoload' option if you don't want augeas to charge all the files/lens, it's 'false' by default.
+# @parameter_constraint noautoload "allow_empty_string" : true
+# @parameter_constraint noautoload "select" : [ "", "true", "false" ]
+# @class_prefix augeas_commands
+# @class_parameter variable_name
+
+
+bundle agent augeas_commands(variable_prefix, variable_name, commands, noautoload)
+{
+ vars:       
+     "old_class_prefix" string => canonify("augeas_commands_${variable_name}");
+     "promisers" slist => { @{this.callers_promisers}, cf_null }, policy => "ifdefined";
+     "class_prefix" string => canonify(join("_", "promisers"));
+     "args" slist => {"${variable_prefix}", "${variable_name}", "${commands}", "${noautoload}"};
+
+   pass1.noautoload_defined::
+     "${variable_prefix}.${variable_name}" string =>  execresult("/usr/bin/env printf \"${commands}\" | augtool --noautoload","useshell"); 
```

Comment:
you use augtool from the $PATH, but you check it with full path

Reply:
you mean the issue is in the 'test' file?